### PR TITLE
Update CHANGELOG for v1.12.1 and v1.12.0

### PR DESCRIPTION
**Pull request by jdn5126**
**What type of PR is this?**
documentation

**Which issue does this PR fix**:
N/A

**What does this PR do / Why do we need it**:
Changelog documentation for v1.12.1 and v1.12.0

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
N/A

**Testing done on this change**:
N/A

**Automation added to e2e**:
N/A

**Will this PR introduce any new dependencies?**:
No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:
No

**Does this PR introduce any user-facing change?**:
No

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,40 @@
 # Changelog
 
+## v1.12.1
+
+* Bug - [Cleanup pod networking resources when IPAMD is unreachable to prevent rule leaking.](https://github.com/aws/amazon-vpc-cni-k8s/pull/2145) (@jdn5126 )
```
Comment (jayanthvn) [inline CHANGELOG.md:5]: Minor nit: let's move the "Bug" below line 15.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+* Improvement - [Move VPC CNI and VPC CNI init images to use EKS minimal base image.](https://github.com/aws/amazon-vpc-cni-k8s/pull/2146) (@jdn5126 )
+* Improvement - [Updating helm chart as per helm v3 standard](https://github.com/aws/amazon-vpc-cni-k8s/pull/2144) (@jaydeokar )
+* Improvement  - [Update golang to 1.19.2](https://github.com/aws/amazon-vpc-cni-k8s/pull/2147) (@jayanthvn )
+* Testing - [Fixes to automation runs](https://github.com/aws/amazon-vpc-cni-k8s/pull/2148) (@jdn5126 )
```
Comment (jayanthvn) [inline CHANGELOG.md:9]: Minor nit : Testing below line 13.

**Review by jayanthvn — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jdn5126) [reply CHANGELOG.md:5]: alphabetized in latest PR

**Review by jayanthvn — APPROVED** [review]
lgtm.

**Pull request merged**